src/render/emitter.ts

```diff
@@ -9,7 +9,11 @@ export class Emitter {
   }
 
   public write (html: any) {
-    if (this.keepOutputType && typeof html !== 'string') {
+    // This will only preserve the type if the value is isolated.
+    // I.E:
+    // {{ my-port }} -> 42
+    // {{ my-host }}:{{ my-port }} -> 'host:42'
+    if (this.keepOutputType === true && typeof html !== 'string' && this.html === '') {
```

Comment:
How about add a separate emitter (like SingleValueEmitter)? It can also check un-intended cases, maybe in the case `{{ my-host }}:{{ my-port }} -> 'host:42'` you need an error thrown?

Reply:
I can create a new Emitter, but I still would like `{{ my-host }}:{{ my-port }} -> 'host:42'` to work without having to specify a specific filter like `| toString`.
Imho, if the value is isolated, it makes sense to keep the original type, if the value is next to others, in most scenario I would say that means you are looking at doing string concatenation.

Reply:
How would you see the new Emitter working? 
If the option `keepOutputType` is `true`, it automatically creates a `SingleValueEmitter` inside `renderTemplates` ? Otherwise, it creates the default HTMLEmitter ?